src/apiRequest/ApiRequest.jsx

```diff
@@ -0,0 +1,130 @@
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {Row, Col} from 'react-flexbox-grid';
+
+import UrlInput from './components/UrlInput';
+import HttpMethodInput from './components/HttpMethodInput';
+import BodyData from './components/BodyData';
+import ApiRequestForm from './components/ApiRequestForm';
+import SuccessToaster from '../components/SuccessToaster';
+import Button from '../components/Button';
+
+import {injectReducer} from '../app/reducers';
+import apiRequestReducer from './ApiRequestReducer';
+import {getTokenId} from '../auth/AuthSelectors';
+import {getApiResponse} from './ApiRequestSelector';
+import {
+  apiRequest,
+  clearApiResponse,
+} from './ApiRequestActions';
+
+export class ApiRequest extends Component {
+  static onEnter(store) {
+    injectReducer(store, {key: 'apiRequestReducer', apiRequestReducer});
+  }
+
+  static defaultProps = {
+    httpMethods: ['GET', 'POST', 'PUT'],
+    apiResponse: undefined,
+    queryList: [],
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      body: '',
+      method: 'GET',
+      query: '',
+    };
+  }
+
+  handleQueryChange = event => this.setState({query: event.target.value})
+
+  handleMethodChange = event => this.setState({method: event.target.value});
+
+  handleBodyChange = code => this.setState({body: code});
+
+  handleSubmit = () => {
+    const {
+      method,
+      body,
+      query,
+    } = this.state;
+
+    const {
+      baseUrl,
+      authToken,
+      makeApiRequest,
+    } = this.props;
+
+    return () => makeApiRequest(method, `${baseUrl}${query}`, body, authToken);
+  }
+
+  render() {
+    const {
+      baseUrl,
+      queryList,
+      httpMethods,
+      apiResponse,
+      clearApiResponse,
+    } = this.props;
+
+    const {
+      body,
+      method,
+      query,
+    } = this.state;
+
+    return (
+      <ApiRequestForm>
+        <UrlInput baseUrl={baseUrl}
+          queryList={queryList}
+          onChange={this.handleQueryChange}
+          query={query}
+        />
+        <HttpMethodInput methods={httpMethods}
+          onChange={this.handleMethodChange}
+          selectedMethod={method}
+        />
+        <BodyData code={body}
+          onChange={this.handleBodyChange}
+        />
+        <Row start="xs">
+          <Col xs={12}>
+            <Button text="Submit"
+              onClick={this.handleSubmit()}
+            />
+            <SuccessToaster title='Success:'
+              onDismiss={clearApiResponse}
+              resultYAML={apiResponse}
+            />
+          </Col>
+        </Row>
+      </ApiRequestForm>
+    );
+  }
+}
+
+const mapStateToProps = state => ({
+  authToken: getTokenId(state),
+  apiResponse: getApiResponse(state),
+});
+
+const mapDispatchToProps = dispatch => ({
+  clearApiResponse: () => dispatch(clearApiResponse()),
+  makeApiRequest: (method, url, body, authToken) => dispatch(apiRequest(method, url, body, authToken)),
+});
```

Comment:
It would be more clear when there will be only handlers to actions without additional dispatchers => actions only in external file :slightly_smiling_face: 
You can use `dispatch` just in actions in mentioned file.